pkg/coscheduling/coscheduling.go

```diff
@@ -71,10 +71,18 @@ func New(ctx context.Context, obj runtime.Object, handle framework.Handle) (fram
 	}
 
 	scheme := runtime.NewScheme()
-	_ = clientscheme.AddToScheme(scheme)
-	_ = v1.AddToScheme(scheme)
 	_ = v1alpha1.AddToScheme(scheme)
-	client, err := client.New(handle.KubeConfig(), client.Options{Scheme: scheme})
+	ccache, err := ctrlcache.New(handle.KubeConfig(), ctrlcache.Options{
+		Scheme: scheme,
+	})
+	if err != nil {
+		return nil, err
+	}
+	go ccache.Start(ctx)
+	ccache.WaitForCacheSync(ctx)
+	client, err := client.New(handle.KubeConfig(), client.Options{
+		Scheme: scheme,
+		Cache:  &client.CacheOptions{Reader: ccache}})
```

Comment:
makes intuitively sense to me but I don't know if the `coscheduling` plugin makes assumption about the freshness of the data, so the cache can have unintended consequences. It would be better to have someone else more familiar with the coscheduling codebase to review.

Reply:
Ok, thanks. I think @denkensk @Huang-Wei are familiar with this plugin. Can you spend some time having a look at this pr?